components/layout_2020/geom.rs

```diff
@@ -730,34 +743,50 @@ impl<T: Clone> Size<T> {
     }
 
     #[inline]
-    pub fn map<U>(&self, f: impl FnOnce(T) -> U) -> Size<U> {
+    pub(crate) fn map<U>(&self, f: impl FnOnce(T) -> U) -> Size<U> {
         match self {
             Size::Initial => Size::Initial,
             Size::MinContent => Size::MinContent,
             Size::MaxContent => Size::MaxContent,
             Size::FitContent => Size::FitContent,
+            Size::FitContentFunction(size) => Size::FitContentFunction(f(size.clone())),
             Size::Stretch => Size::Stretch,
             Size::Numeric(numeric) => Size::Numeric(f(numeric.clone())),
         }
     }
 
     #[inline]
-    pub fn maybe_map<U>(&self, f: impl FnOnce(T) -> Option<U>) -> Option<Size<U>> {
-        Some(match self {
-            Size::Numeric(numeric) => Size::Numeric(f(numeric.clone())?),
+    pub(crate) fn maybe_map_for_preferred<U>(&self, f: impl FnOnce(T) -> Option<U>) -> Size<U> {
+        match self {
+            Size::Numeric(numeric) => f(numeric.clone()).map_or(Size::Initial, Size::Numeric),
+            Size::FitContentFunction(numeric) => {
+                f(numeric.clone()).map_or(Size::FitContent, Size::FitContentFunction)
+            },
             _ => self.map(|_| unreachable!("This shouldn't be called for keywords")),
-        })
+        }
+    }
+
+    #[inline]
+    pub(crate) fn maybe_map_for_max<U>(&self, f: impl FnOnce(T) -> Option<U>) -> Size<U> {
```

Comment:
Some documentation on these new methods would be "nice to have" (e.g. what are used they for)

Reply:
Trying to write some documentation made me realize that these were too tied to percentage resolution and weren't making much sense in `Size<T>`, so I have removed them and instead added `resolve_percentages_for_preferred()` and `resolve_percentages_for_max()` in `Size<LengthPercentage>`.